core_data_modules/traced_data/io.py

```diff
@@ -309,18 +309,19 @@ def import_coda_to_traced_data_iterable(user, data, key_of_raw, scheme_keys, f,
         # TODO: Test when running on a machine set to German.
         imported_csv = csv.DictReader(f, delimiter=";")
 
-        # Remove rows which still haven't been coded.
-        coded = list(filter(lambda row: row["deco_codeValue"] != "", imported_csv))
+        # Build a lookup table of (raw_data, schemeName) -> row for only the rows which have been coded.
+        coded = {(row["data"], row["schemeName"]): row for row in imported_csv if row["deco_codeValue"] != ""}
 
         for td in data:
             for scheme_name, key_of_coded in scheme_keys.items():
                 if not overwrite_existing_codes and td.get(key_of_coded) is not None:
                     continue
 
-                code = None
-                for row in coded:
-                    if td[key_of_raw] == row["data"] and row["schemeName"] == scheme_name:
-                        code = row["deco_codeValue"]
+                coded_lookup_key = (td[key_of_raw], scheme_name)
+                if coded_lookup_key in coded:
+                    code = coded[coded_lookup_key]["deco_codeValue"]
```

Comment:
driveby comment unrelated to this PR - could we pull the magic constants out to the top of the function?